tuning/data/data_handlers.py

```diff
@@ -0,0 +1,83 @@
+# Copyright The FMS HF Tuning Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Definition of some predefined data preprocessing functions that we need.
+
+# Standard
+from typing import Dict, List
+
+# Third Party
+from transformers import AutoTokenizer
+
+# Local
+from tuning.utils.data_utils import custom_data_formatter
+from tuning.utils.preprocessing_utils import combine_sequence
+
+
+def tokenize_and_apply_instruction_masking(
+    element: Dict[str, str],
+    tokenizer: AutoTokenizer,
+    input_field_name: str,
+    output_field_name: str,
+    **tokenizer_kwargs,
+):
+    input = element[input_field_name]
+    output = element[output_field_name]
+
+    # TODO: Eventually move the code here
+    combined = combine_sequence(input, output, eos_token=tokenizer.eos_token)
+
+    tokenized_comb_seqs = tokenizer(combined, **tokenizer_kwargs)
+    tokenized_input = tokenizer(input, **tokenizer_kwargs)
+
+    masked_labels = [-100] * len(
+        tokenized_input.input_ids
+    ) + tokenized_comb_seqs.input_ids[len(tokenized_input.input_ids) :]
+
+    # Any benefit of retaining the old columns?
+    return {
+        "input_ids": tokenized_comb_seqs.input_ids,
+        "labels": masked_labels,
+        "attention_mask": tokenized_comb_seqs.attention_mask,
+    }
+
+
+def apply_dataset_formatting(
+    element: Dict[str, str], tokenizer: AutoTokenizer, dataset_text_field: str, **kwargs
+):
+    return {
+        f"{dataset_text_field}": element[f"{dataset_text_field}"] + tokenizer.eos_token
+    }
+
```

Comment:
When [raw_datasets = raw_datasets.map(handler, **kwargs)](https://github.com/foundation-model-stack/fms-hf-tuning/blob/5e948f16a2c7fa4d942af434917d5483538827a0/tuning/data/data_processors.py#L166) is called and `kwargs["batched"] = True` then `element[f"{dataset_text_field}"]` here would be a list right. So does before condition make sense:

```
if "batched" in kwargs and kwargs["batched"]:
      return {
          f"{dataset_text_field}": element[f"{dataset_text_field}"] + tokenizer.eos_token
      }
  return {
      f"{dataset_text_field}": [
          text + tokenizer.eos_token for text in element[f"{dataset_text_field}"]
      ]
  }
```

Reply:
@Abhishek-TAMU I consciously reverted this change and made our default handlers run in [`batched=False`](https://github.com/dushyantbehl/fms-hf-tuning/blob/96b94676038fe6206ef09ea192613f21a1a66b83/tuning/data/setup_dataprocessor.py#L94) mode for now, we can mention this in documentation for people to not use these handlers in `batched` mode. This change was made due to the fact that this and the other handlers we have defined right now are complex operations and need us to deconstruct each example from a batch before processing them. 

That being said this was after I had already implemented a patch to make all handlers take in `batched` and `non-batched` input and it is here in a [different branch](https://github.com/dushyantbehl/fms-hf-tuning/commit/ef2ef7808e547946bc395ce7cd83d1b71197c2b7)
so we can even cherry pick this to the current branch if needed. 

cc @Ssukriti @willmj @ashokponkumar

Reply:
> so we can even cherry pick this to the current branch if needed.

This sounds fine to me.